clang/test/CodeGenSYCL/kernel-arg-accessor-pointer.cpp

```diff
@@ -0,0 +1,122 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
+
+// This test checks if the metadata "kernel-arg-runtime-aligned"
+// is generated if the kernel captures an accessor
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+
+queue q;
+
+int main() {
+
+  using Accessor =
+      accessor<int, 1, access::mode::read_write, access::target::global_buffer>;
+  Accessor acc[2];
+
+  accessor<int, 1, access::mode::read, access::target::global_buffer> readOnlyAccessor;
+
+  accessor<float, 2, access::mode::write,
+           access::target::local,
+           access::placeholder::true_t>
+      acc3;
+
+  // kernel_A parameters : int*, sycl::range<1>, sycl::range<1>, sycl::id<1>,
+  // int*, sycl::range<1>, sycl::range<1>,sycl::id<1>.
+  q.submit([&](handler &h) {
+    h.single_task<class kernel_A>([=]() {
+      acc[1].use();
+    });
+  });
+
+  // kernel_readOnlyAcc parameters : int*, sycl::range<1>, sycl::range<1>, sycl::id<1>.
+  q.submit([&](handler &h) {
```

Comment:
Can you add a test where the kernel captures an accessor and something else - maybe a raw pointer and an int.

Reply:
I have included a test case that captures an accessor as well as a raw pointer.
We have a test case for each of them individually.
Could you explain the value/motivation for having another test case with all of them combined?

Reply:
Just to have a slightly more complex test capturing multiple variables as opposed to just one.